sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/LiteralGenerator.scala

```diff
@@ -78,7 +78,17 @@ object LiteralGenerator {
         Double.NaN, Double.PositiveInfinity, Double.NegativeInfinity)
     } yield Literal.create(f, DoubleType)
 
-  // TODO: decimal type
+
+  // TODO cache the generated data, and boarder the value range
+  // Since scalacheck doesn't support the decimal yet in `Choose`.
+  def decimalLiteralGen(precision: Int, scale: Int): Gen[Literal] = {
+    Gen.chooseNum(Double.MinValue, Double.MaxValue,
+      Double.NaN, Double.PositiveInfinity, Double.NegativeInfinity) map { bd =>
+      val d = Decimal(bd)
+      d.changePrecision(precision, scale)
```

Comment:
Yes, that's true, I will udpate the code by gen the random decimal according to the `precision` and `scale`.